manifests/charts/istio-control/istio-discovery/files/injection-template.yaml

```diff
@@ -35,6 +35,7 @@ metadata:
     service.istio.io/canonical-name: {{ index .ObjectMeta.Labels `service.istio.io/canonical-name` | default (index .ObjectMeta.Labels `app.kubernetes.io/name`) | default (index .ObjectMeta.Labels `app`) | default .DeploymentMeta.Name  | quote }}
```

Comment:
we have like 7 templates in this folder now, can we update them all? this will ensure waypoints,injected gateways, etc get it.

(it would be nice if we had some easy way to have common parts between them all but that is complex and way outside of scope of this PR)